### PR TITLE
WIP - Gatsby setup for documentation website

### PR DESCRIPTION
**Pull request by emmenko**
**Still in progress**

screenshot 2019-01-30 at 22 29 53
screenshot 2019-01-30 at 22 29 57
screenshot 2019-01-30 at 22 30 31

---

Scope:
- [x] setup minimal gatsby project
- [x] add mdx support
- [x] render ui-kit components
- [x] add syntax highlighting
- [x] create pages based on mdx component files
- [x] document basic setup
- [x] basic scaffolding layout
- [x] try react-live for live editor/preview
- [x] add basic routes
- [ ] configure navigation menu
- [ ] color palette / theme?
- [ ] clean it up and make it ready for merge

### CODE REVIEW

File: website/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      // }
+      // Unless you're replacing Babel with a different transpiler, you probably
+      // want this so that Gatsby will apply its required Babel
+      // presets/plugins.  This will also merge in your configuration from
+      // `babel.config.js`.
+      ...loaders.js(),
+
+      test: /\.jsx?$/,
+
+      // Exclude all node_modules from transpilation, except for 'swiper' and 'dom7'
```
Comment (montezume) [inline website/gatsby-node.js]: 🤔

File: website/gatsby-node.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      exclude: [/(node_modules)/, /(ui-kit.esm)/],
+    },
+  ];
+
+  config.plugins = [
+    ...config.plugins,
+    // See https://github.com/FormidableLabs/react-live/issues/5
```
Comment (montezume) [inline website/gatsby-node.js]: Do we need this? That issue was closed last year

**Comment by montezume** [comment]
I'm also wondering if the live editing with react-live is a core requirement?

I like how shopify does their doc site and they have this feature, and I think it's fine without it.

https://polaris.shopify.com/components/get-started

**Comment by montezume** [comment]
Is the react-live a core requirement of the website? I sort of like how material-ui doesn't have this and just has sandbox links

https://material-ui.com/demos/tooltips/

**Comment by emmenko** [comment]
I'm just experimenting with it to see if it makes sense to use it or not. For instance, the react website uses it.

Everything is still up to debate

File: src/components/fields/date-time-field/date-time-field.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  <DateTimeField
+    title="Release Date"
+    value="2018-11-30T13:25:59.500Z"
+    timeZone="Europe/Berlin"
+    onChange={event => alert(event.target.value)}
+  />
+</IntlProvider>
+```
+
+#### Properties
```
Comment (montezume) [inline src/components/fields/date-time-field/date-time-field.mdx]: having props duplicated here and in the README is another reason why we should generate them. 🤔

Comment (emmenko) [reply src/components/fields/date-time-field/date-time-field.mdx]: Yeah sure. Btw, this was basically just a dump of the readme 😅

**Comment by montezume** [comment]
I like the header 👏

**Comment by qmateub** [comment]
Wow! This is looking awesome Nicola! Looking forward for it once you are done 👏

**Comment by emmenko** [comment]
Closing for now

**Pull request closed without merging**